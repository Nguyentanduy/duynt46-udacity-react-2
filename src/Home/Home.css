.border-box{
    display: grid;
    grid-template-columns: auto auto auto auto;
    width: 80%;
    margin: 15px auto;
    border: 1px solid gray;
    padding: 0px 20px 20px 20px;
    border-radius: 5px;
}
.home{
    text-align: center;
}