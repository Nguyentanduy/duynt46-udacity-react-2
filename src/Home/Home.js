import React, { useEffect, useState } from 'react'
import NavBar from '../components/NavBar/NavBar'
import Card from '../components/Card/Card'
import './Home.css'
import { setAllQuestions } from '../Slices/QuestionsSlices'
import { useDispatch, useSelector } from 'react-redux'
import { _getQuestions } from '../_DATA'
export default function Home() {
    const dispatch = useDispatch()
    const [newQues, setNewQues] = useState()
    const [doneQues, setDoneQues] = useState()
    const [isQues, setIsQues] = useState(true)
    const allQuestions = useSelector(state => state.questions.allQuestions)
    const user = useSelector((state) => state.user.user);

    useEffect(() => {
        _getQuestions().then(data => {
            dispatch(setAllQuestions(data))
        })
    }, [])

    useEffect(() => {
        if (allQuestions && user.answers) {
            let questionList = Object.values(allQuestions);
            let answersUser = Object.keys(user?.answers);
            setNewQues(questionList.filter(
                (item) => !answersUser.includes(item.id)
            ))
            setDoneQues(questionList.filter(
                (item) => answersUser.includes(item.id)
            ))
        }
    }, [allQuestions])

    const handleSwithPoll = (isPoll) => {
        setIsQues(isPoll)
    }

    return (
        <div>
            <NavBar />
            <div className='home'>
                <div style={{ display: 'flex' }}>
                    <div className='button' onClick={() => handleSwithPoll(true)}>New Questions</div>
                    <div className='button' onClick={() => handleSwithPoll(false)}>Done</div>
                </div>
                {isQues ? <div>
                    <div style={{ fontSize: "20px", fontWeight: '600' }}>New Questions</div>
                    <div className='border-box'>
                        {newQues?.map((item) => {
                            return <Card key={item.id} item={item} />
                        })}
                    </div>
                </div> : <div>
                    <div style={{ fontSize: "20px", fontWeight: '600' }}>Done</div>
                    <div className='border-box'>
                        {doneQues?.map((item) => {
                            return <Card key={item.id} item={item} />
                        })}
                    </div>
                </div>}


            </div>
        </div>
    )
}
