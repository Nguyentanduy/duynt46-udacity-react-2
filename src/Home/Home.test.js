import { render } from "@testing-library/react";
import * as React from 'react';
import Home from "./Home";

describe('Home',() => {
    it('will match snapshot', () => {
        var component = render(<Home/>);
        expect(component).toMatchSnapshot();
    })
})