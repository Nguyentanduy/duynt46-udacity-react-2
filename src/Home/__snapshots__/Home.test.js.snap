// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home should be match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navbar_container"
        >
          <div
            style="display: flex;"
          >
            <div
              class="button"
            >
              Home
            </div>
            <div
              class="button"
            >
              Leaderboard
            </div>
            <div
              class="button"
            >
              New
            </div>
          </div>
          <div
            style="display: flex;"
          >
            <img
              alt="user"
              height="40"
              src="user1.png"
              width="70"
            />
            <div
              class="button"
            />
            <div
              class="button"
            >
              Logout
            </div>
          </div>
        </div>
        <hr />
        <div
          class="home"
        >
          <div>
            <div
              style="font-size: 20px; font-weight: 600;"
            >
              New Questions
            </div>
            <div
              class="border-box"
            />
          </div>
          <div>
            <div
              style="font-size: 20px; font-weight: 600;"
            >
              Done
            </div>
            <div
              class="border-box"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navbar_container"
      >
        <div
          style="display: flex;"
        >
          <div
            class="button"
          >
            Home
          </div>
          <div
            class="button"
          >
            Leaderboard
          </div>
          <div
            class="button"
          >
            New
          </div>
        </div>
        <div
          style="display: flex;"
        >
          <img
            alt="user"
            height="40"
            src="user1.png"
            width="70"
          />
          <div
            class="button"
          />
          <div
            class="button"
          >
            Logout
          </div>
        </div>
      </div>
      <hr />
      <div
        class="home"
      >
        <div>
          <div
            style="font-size: 20px; font-weight: 600;"
          >
            New Questions
          </div>
          <div
            class="border-box"
          />
        </div>
        <div>
          <div
            style="font-size: 20px; font-weight: 600;"
          >
            Done
          </div>
          <div
            class="border-box"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navbar_container"
        >
          <div
            style="display: flex;"
          >
            <div
              class="button"
            >
              Home
            </div>
            <div
              class="button"
            >
              Leaderboard
            </div>
            <div
              class="button"
            >
              New
            </div>
          </div>
          <div
            style="display: flex;"
          >
            <img
              alt="user"
              height="40"
              src="user1.png"
              width="70"
            />
            <div
              class="button"
            />
            <div
              class="button"
            >
              Logout
            </div>
          </div>
        </div>
        <hr />
        <div
          class="home"
        >
          <div>
            <div
              style="font-size: 20px; font-weight: 600;"
            >
              New Questions
            </div>
            <div
              class="border-box"
            />
          </div>
          <div>
            <div
              style="font-size: 20px; font-weight: 600;"
            >
              Done
            </div>
            <div
              class="border-box"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navbar_container"
      >
        <div
          style="display: flex;"
        >
          <div
            class="button"
          >
            Home
          </div>
          <div
            class="button"
          >
            Leaderboard
          </div>
          <div
            class="button"
          >
            New
          </div>
        </div>
        <div
          style="display: flex;"
        >
          <img
            alt="user"
            height="40"
            src="user1.png"
            width="70"
          />
          <div
            class="button"
          />
          <div
            class="button"
          >
            Logout
          </div>
        </div>
      </div>
      <hr />
      <div
        class="home"
      >
        <div>
          <div
            style="font-size: 20px; font-weight: 600;"
          >
            New Questions
          </div>
          <div
            class="border-box"
          />
        </div>
        <div>
          <div
            style="font-size: 20px; font-weight: 600;"
          >
            Done
          </div>
          <div
            class="border-box"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
